packages/algorand-js/src/interpreter/opcode-list.ts

```diff
@@ -151,3 +157,319 @@ export class Intc extends Op {
+    this.assertStackLen(stack, 2);
+    const a = this.assertBigInt(stack.pop());
+    const b = this.assertBigInt(stack.pop());
+    if (a === b) {
+      stack.push(BigInt('1'));
+    } else {
+      stack.push(BigInt('0'));
+    }
+  }
+}
+
+export class NotEqualTo extends Op {
+  execute (stack: TEALStack): void {
+    this.assertStackLen(stack, 2);
+    const a = this.assertBigInt(stack.pop());
+    const b = this.assertBigInt(stack.pop());
+    if (a !== b) {
+      stack.push(BigInt('1'));
+    } else {
+      stack.push(BigInt('0'));
+    }
+  }
+}
+
+export class Not extends Op {
+  execute (stack: TEALStack): void {
+    const a = this.assertBigInt(stack.pop());
+    if (a === BigInt('0')) {
+      stack.push(BigInt('1'));
+    } else {
+      stack.push(BigInt('0'));
+    }
+  }
+}
+
+// converts uint64 X to big endian bytes
+export class Itob extends Op {
+  execute (stack: TEALStack): void {
+    const stackValue = stack.pop();
+    const buf = encode(Number(stackValue));
+    const uint8arr = new Uint8Array(buf);
+    stack.push(uint8arr);
+  }
+}
+
+// converts bytes X as big endian to uint64
+// btoi panics if the input is longer than 8 bytes.
+export class Btoi extends Op {
+  execute (stack: TEALStack): void {
+    const uint8arr = stack.pop() as Uint8Array;
+    if (uint8arr.length > 8) {
+      throw new Error("Input is longer than 8 bytes");
+    }
+    const buf = Buffer.from(uint8arr);
+    const uintValue = decode(buf);
+    stack.push(uintValue);
+  }
+}
+
+export class Mulw extends Op {
+  execute (stack: TEALStack): void {
+    // To-DO
+  }
+}
+
+// A plus B out to 128-bit long result as sum (top) and carry-bit uint64 values on the stack
+export class Addw extends Op {
+  execute (stack: TEALStack): void {
+    const valueA = stack.pop() as bigint;
+    const valueB = stack.pop() as bigint;
+    let valueC = valueA + valueB;
+
+    if(valueC > MAX_UINT64) {
+      valueC -= MAX_UINT64;
+      stack.push(BigInt('1'));
+      stack.push(valueC - BigInt('1'));
+    } else {
+      stack.push(BigInt('0'));
+      stack.push(valueC);
+    }
+  }
+}
+
+// store fields and get them from current context
+export class txn extends Op {
+  execute (stack: TEALStack): void {
+    // txn CloseRemainderTo
+    // push closeRemainderTo from transaction data
+  }
+}
+
+export class Pop extends Op {
+  execute (stack: TEALStack): void {
+    stack.pop();
+  }
+}
+
+// duplicate last value on stack
+export class Dup extends Op {
+  execute (stack: TEALStack): void {
+    const lastValue = stack.pop() as bigint;
+   
+    stack.push(lastValue);
+    stack.push(lastValue);
+  }
+}
+
+// duplicate two last values on stack: A, B -> A, B, A, B
+export class Dup2 extends Op {
+  execute (stack: TEALStack): void {
+    const lastValueA = stack.pop() as bigint;
+    const lastValueB = stack.pop() as bigint;
+    
+    stack.push(lastValueA);
+    stack.push(lastValueB);
+    stack.push(lastValueA);
+    stack.push(lastValueB);
```

Comment:
- Same as `Dup`
- Also, can you confirm the logic here? I looks like before popping top of stack was A, and after pushing all 4 values top of stack is B.